### ISSUE TITLE
buffer fully zero after the mp3dec_decode_frame() call

### ISSUE THREAD

**Issue by siladrenja** [lieff/minimp3#105]
Sorry to bother you, I don't know what I'm doing wrong, but when I tried to integrate it with my audio playback library, I get the correct output in everything except the buffer bits. Samples sys 1152, and read_bytes says 1181 (which does gravitate around it a little, but pretty much in that range)

I'm testing it on a "Hail to the king" quote from the duke nukem that I found on the internet (the file is called "HailKing.mp3" as can be seen in the code, in VLC and also windows media player it plays correctly, so it is a valid file)

Here's my code:
```cpp
#include "AudioPlayer.hpp"
#include "queuePointer.h"
#include <fstream>
#define MINIMP3_IMPLEMENTATION
#include "minimp3.h"
#include <iostream>

using namespace AP;



int wmain(int Argc, wchar_t** Argv) {


    const wchar_t* FileName = L"HailKing.mp3";
    
    std::ifstream inputFile(FileName, std::ios::binary);

    char InBuff[16*1024];
    size_t BufferCounter = 16*1024;
    inputFile.read(InBuff, 16 * 1024);

    unsigned char* InBuffIter = (unsigned char*)InBuff;
    static mp3dec_t mp3d;
    mp3dec_init(&mp3d);
    

    WAVEFORMATEX wfx;
    wfx.wFormatTag = WAVE_FORMAT_PCM;
    wfx.nChannels = 2;
    wfx.nSamplesPerSec = 44100;
    wfx.nAvgBytesPerSec = wfx.nSamplesPerSec * wfx.nChannels * 2;
    wfx.nBlockAlign = wfx.nChannels * 2;
    wfx.wBitsPerSample = 16;
    
    AudioPlayer ap(wfx, MINIMP3_MAX_SAMPLES_PER_FRAME * 4, [&](char* buff, size_t buffSize) {
        mp3dec_frame_info_t info;
        
        int samples = mp3dec_decode_frame(&mp3d, InBuffIter, BufferCounter, (short*)buff, &info);
        InBuffIter += samples;
        BufferCounter -= samples;
        if (BufferCounter <= 3 * MINIMP3_MAX_SAMPLES_PER_FRAME) {
            std::copy(InBuff + (16 * 1024 - BufferCounter), InBuff + 16 * 1024, InBuff);
            size_t bytesRead = inputFile.read(InBuff + BufferCounter, 16 * 1024 - BufferCounter).gcount();
            InBuffIter = (unsigned char*)InBuff;
            BufferCounter = 16 * 1024 - bytesRead;
        }
        std::cout << "Decoded " << samples << " samples (" << samples * 4 << " bytes)" << std::endl;
        std::cout << "Read " << info.frame_bytes << " bytes from input buffer" << std::endl;

        for (int i = 0; i < 4096; i++) {
            std::cout << +buff[i] << " ";
        }
        std::cout << '\n';

        
        return samples * 4;
    });

    ap.FinalLoop();
    //system("pause");
}
```

Explanation of what I'm doing:
So basically, minimp3 is mainly being used in the lambda passed to the constructor of AudioPlayer object. char* buff in it's arguments is the buffer itself, and buffSize is maximum (aka allocated) size of the buffer.

**Comment by siladrenja** [comment]
I rewrote the code and suddenly it works. IDK what exactly was the problem, but turns out the problem was with my code.
